src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -271,8 +271,18 @@ private void ReadSettings(PackageRestoreInputs packageRestoreInputs)
                     GetInstalledPackageReferences(packageReferenceFile, allowDuplicatePackageIds: true));
             }
 
+            Func<Packaging.Core.PackageIdentity, bool> packageExistsFunc;
+            // If restoring nupkg only accept an existing package with a nupkg otherwise accept a nuspec or a nupkg
+            if ((EffectivePackageSaveMode & Packaging.PackageSaveMode.Nupkg) == Packaging.PackageSaveMode.Nupkg)
+            {
+                packageExistsFunc = nuGetPackageManager.PackageExistsInPackagesFolder;
+            }
+            else
+            {
+                packageExistsFunc = nuGetPackageManager.PackageOrManifestExistsInPackagesFolder;
```

Comment:
If `PackageSaveMode == Nuspec`, then `packageExistsFunc` could return true if the .nupkg exists but the .nuspec does not exist right?